packages/mwp-core/src/renderers/server-render.jsx

```diff
@@ -279,15 +279,18 @@ const makeRenderer = (
 			state.preRenderChecklist.every(isReady => isReady);
 		const populateStore = store =>
 			new Promise((resolve, reject) => {
-				// dispatch SERVER_RENDER to kick off API middleware
-				store.dispatch({ type: SERVER_RENDER, payload: url });
-
+				// keep this above the SERVER_RENDER so
+				// state.flags is available to query calls
 				if (checkReady(store.getState())) {
 					addFlags(store).then(() => {
 						resolve(store);
 					});
 					return;
 				}
+
+				// dispatch SERVER_RENDER to kick off API middleware
+				store.dispatch({ type: SERVER_RENDER, payload: url });
```

Comment:
Isn't is possible that `checkReady` would not be true and that `SERVER_RENDER` might still dispatch before `addFlags`? I think we may just have to move this beneath `resolve(store` in the `addFlags` resolution, unless there's an explicit reason we should not or don't have to.